### DISCUSSION TITLE
De-Graining

### DISCUSSION THREAD

**Discussion by graciepii** [casact/chainladder-python#527] [Q&A]
Hello,

I have a set of calculations that iterates back through triangles by year. When I change the grain of the triangle it then iterates by Month / Quarter which is not ideal. Is there a way to switch from say OYDQ to OYDY, and then after performing calculations back to OYDQ? I think this is not possible, but it would be a seriously handy feature to have!

**Comment by jbogaardt** [comment]
Unfortunately,
No. Grain is an aggregation method like sum. Once applied, the details are lost and you cannot go from a higher grain down to a lower grain.

You could make copies of the lower grain triangle and maintain them for future use using the copy method.
